commands/01-mod.js

```diff
@@ -1104,7 +1104,9 @@ Module
   try {
     // Pin Request
     if (message.guild && (message.guild.id == Module.config.ldsg) && (reaction.emoji.name == "📌") && message.pinnable) {
-      if (message.channel.permissionsFor(user).has("MANAGE_MESSAGES")) message.pin();
+      if (message.channel.permissionsFor(user).has("MANAGE_MESSAGES")){
+	      message.channel.messages.fetchPinned().then(m =>{if(m.array().length == 50) return message.channel.send("I was unable to pin the message since the channel pin limit has been reached."); else message.pin();})
+      }
```

Comment:
Use `await` functionality to keep things clean.
```suggestion
      if (message.channel.permissionsFor(user).has("MANAGE_MESSAGES")){
        let messages = await message.channel.messages.fetchPinned().catch(u.noop);
        if (messages?.size == 50) return message.channel.send(user + ", I was unable to pin the message since the channel pin limit has been reached.");
        else message.pin();
      }
```